tools/cp-caps/rcp_caps_test.py

```diff
@@ -0,0 +1,414 @@
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import argparse
+import logging
+import os
+import sys
+import textwrap
+
+from typing import List
+
+import otci
+from otci import OTCI
+
+logging.basicConfig(level=logging.WARNING)
+
+
+class RcpCaps(object):
+    """
+    This class represents an OpenThread RCP capability test instance.
+    """
+
+    def __init__(self):
+        self.__dut = self.__connect_dut()
+        self.__ref = self.__connect_reference_device()
+
+    def test_diag_commands(self):
+        """Test all diag commands."""
+        self.__dut.factory_reset()
+        self.__ref.factory_reset()
+
+        ret = self.__dut.is_command_supported('diag start')
+        if ret is False:
+            print('All diag commands are not supported')
+            return
+
+        self.__dut.diag_start()
+        self.__ref.diag_start()
+
+        self.__test_diag_channel()
+        self.__test_diag_power()
+        self.__test_diag_radio()
+        self.__test_diag_repeat()
+        self.__test_diag_send()
+        self.__test_diag_frame()
+        self.__test_diag_echo()
+        self.__test_diag_utils()
+        self.__test_diag_rawpowersetting()
+        self.__test_diag_powersettings()
+        self.__test_diag_gpio_mode()
+        self.__test_diag_gpio_value()
+
+        self.__ref.diag_stop()
+        self.__dut.diag_stop()
+
+    #
+    # Private methods
+    #
+    def __test_diag_channel(self):
+        channel = 20
+        commands = ['diag channel', f'diag channel {channel}']
+
+        if self.__support_commands(commands):
+            self.__dut.diag_set_channel(channel)
+            channel = self.__dut.diag_get_channel()
+            ret = True if channel == channel else False
```

Comment:
I don't understand this check. Is here a bug?

Reply:
I have fixed this bug